go/chat/boxer.go

```diff
@@ -89,38 +89,45 @@ func (b *Boxer) makeErrorMessage(msg chat1.MessageBoxed, err UnboxingError) chat
 	})
 }
 
-func (b *Boxer) detectKBFSPermanentServerError(err error) bool {
+func (b *Boxer) detectKBFSPermanentServerError(err error, tlfName string) UnboxingError {
 	// Banned folders are only detectable by the error string currently, hopefully
 	// we can do something better in the future.
 	if err.Error() == "Operations for this folder are temporarily throttled (error 2800)" {
-		return true
+		return NewPermanentUnboxingError(err)
 	}
 
 	// Check for team not exist error that is in raw form
 	if aerr, ok := err.(libkb.AppStatusError); ok {
 		switch keybase1.StatusCode(aerr.Code) {
 		case keybase1.StatusCode_SCTeamNotFound:
-			return true
+			return NewPermanentUnboxingError(err)
 		case keybase1.StatusCode_SCTeamReadError:
 			// These errors get obfuscated by the server on purpose. Just mark this as permanent error
 			// since it likely means the team is in bad shape.
-			return aerr.Error() == "You are not a member of this team (error 2623)"
+			if aerr.Error() == "You are not a member of this team (error 2623)" {
+				return NewPermanentUnboxingError(err)
+			}
+			return NewTransientUnboxingError(err)
 		}
 	}
 
-	switch err.(type) {
-	case libkb.DeletedError:
-		return true
+	switch err := err.(type) {
+	case libkb.UserDeletedError:
```

Comment:
Does `DeletedError` not happen anymoer?

Reply:
I renamed it. It was only being used for when a user is deleted. And I thought maybe the name was part of why the messages were so vague.

Also I'm working on making the gui message better cause it's still bad.

Reply:
Ah ok, I should have scrolled down further.